WalletWasabi.Tests/UnitTests/JsonConverters/MoneySatoshiJsonConverterTests.cs

```diff
@@ -0,0 +1,196 @@
+
+namespace WalletWasabi.Tests.UnitTests.JsonConverters;
+
+public class MoneySatoshiJsonConverterTests
+{
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>Newtonsoft.Json</c> *serialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void SerializationParity()
+	{
+		TestData testObject = new();
+
+		string json = AssertSerializedEqually(testObject);
+		Assert.Equal("""{"OneCoin":100000000,"OneSatoshi":1,"SmallAmount":100,"Zeros":0,"Max":2099999997690000,"None":null,"NotAnnotated":null}""", json);
+	}
+
+	/// <summary>
+	/// Tests that JSON converter based on <c>System.Text.Json</c> and the one based on <c>Newtonsoft.Json</c> *deserialize* objects equally.
+	/// </summary>
+	[Fact]
+	public void DeserializationParity()
+	{
+		// Success cases.
+		{
+			string token = "2100000000000000"; // 21 million.
+			AssertBothDeserialize(token);
+		}
+
+		// Json exceptions.
+		// Newtonsoft gives JsonReaderException, Microsoft gives JsonException. Similar but not the same.
+		{
+			string invalidToken = "1,000.00"; // Thousand separator.
+			AssertDeserializeJsonException(invalidToken);
+
+			invalidToken = "1,0"; // Decimal comma.
+			AssertDeserializeJsonException(invalidToken);
+		}
+
+		// Casting errors Vs Format Exception.
+		{
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			string invalidToken = "0.00000000000000000000000000000000000000000000001";
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			invalidToken = "209999999.97690001"; // Maximum number of bitcoins ever to exist + 1 satoshi.
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back FormatException.
+			invalidToken = "1e6"; // Exponential notation.
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+		}
+
+		// Unique case.
+		{
+			// Newtonsoft gives back InvalidCastException, Microsoft gives back JsonException (Read function not even called).
+			string invalidToken = "1."; // No digit after decimal point.
+			AssertDeserializeDifferentExceptions<InvalidCastException, System.Text.Json.JsonException>(invalidToken);
+
+			// TODO: remove https://stackoverflow.com/questions/27361565/why-is-json-invalid-if-an-integer-begins-with-a-leading-zero
+			// https://www.rfc-editor.org/rfc/rfc4627.txt - section 2.4. Numbers mentions "Leading zeros are not allowed."
+			// Newtonsoft reads 0, Microsoft gives back JsonException.
+			// token = "00000000000000000000000";
+			// AssertBothDeserialize(token);
+		}
+
+		// Tests that neither JSON converters can deserialize to NULL if a JSON number-string is found instead of a JSON integer.
```

Comment:
I'm not sure if I understand this comment. 
`neither JSON converters can deserialize to NULL` to NULL?